pyvista/plotting/plotting.py

```diff
@@ -6860,6 +6860,75 @@ def add_cursor(
 
         return actor
 
+    def add_dimension_line(self, pointa, pointb, direction, **kwargs):
+        """Add a dimension line of a PyVista or VTK dataset to the scene.
+
+        Parameters
+        ----------
+        pointa : sequence[float]
+            Length 3 coordinate of the start of the line.
+
+        pointb : sequence[float]
+            Length 3 coordinate of the end of the line.
+
+        direction : array_like[float]
+            Direction vector for dimension line.
+
+        **kwargs : dict, optional
```

Comment:
I now think it would be better to screen the available arguments. I would like to remove them.

Reply:
Fixed in https://github.com/pyvista/pyvista/pull/4285/commits/399e7bf87f152d66c699f5c51b0feda9b4102504 .

Reply:
Perhaps I am not understanding your point correctly due to my ignorance. Could you please be more specific?

Reply:
I like that you now have named arguments that are specific to each use.  But I'm wondering if we will end up with an explosion of options here.  For example, I can imagine we could add options to control the length, thickness, and color of the connecting line that points from the dimension line to the starting points.  The user may want to have control over the thickness or color of the dimension lines.  Or, the user may want to use arrows for the dimension line instead of just a line.  I'm trying to have a discussion on what the best user interface looks like.  

Will we continue to add named keyword parameters?  In this case, we should use keyword only arguments to ensure that we can group parameters in the future without breaking changes.  Should we group parameters into a dict?  In this case, we could have a parameter for the dimension line and then pass all of them into the `add_mesh` call, for example it would look like


```py
dimension_line_opyptions = {"color": "red"}
pl.add_dimension_line(pointa, pointb, offset, dimension_line_options)
```

Inside `add_dimension_line`, we would do:

```py
        _ = self.add_mesh(arrows, **dimension_line_options)
```

I don't I have a strong opinion in which direction (and there may be more directions), but we should think it through so we don't end up having pain later.  The dict option benefit is that we won't have to change this function if we add options in add_mesh.

Reply:
> Will we continue to add named keyword parameters?

It is already a pain to add parameters 🥲 I would like to follow your suggestion.
I would like to also hear @pyvista/developers 's opinion on which direction is better in this case.